controllers/todo.js

```diff
@@ -1,33 +1,247 @@
 const { ToDo, Token } = require("../models");
+const todo = require("../models/todo");
 
 // All the given method require token.
 // So be sure to check for it before doing any stuff
 // HINT: Create a middleware for above :)
 
+//ok lets create a middleware XD
+const verification = (req, res) => {
+  try{
+    const token = req.body.token || req.query.token ||
+                  req.headers["x-access-token"] || req.headers.token ||
+                  req.headers.authorization.split(' ')[1];
+    if (!token)
+      return false;
+    else
+      return token;
+  }
+  catch(err){
+    return false;
+  }
+};
 const getAllToDo = async (req, res) => {
   // Get the token in header.
   // Use the token to get all the ToDo's of a user
+  const token = verification(req,res);
+ if(token != false){
+   Token.findOne({token: token}, function(err, foundToken){
+     if(err){
+       res.send(err);
+       res.send("hello");
+     }
+     else{
+       if(foundToken){
+             ToDo.find({ $or: [{createdBy : foundToken.user},
+                               {collaborators: {"$in" : foundToken.user._id.toString()}}
+                             ]}, function(err, foundUsers){
+               if(foundUsers){
+                 users = [];
+                 foundUsers.forEach(function(user){
+                   users.push({
+                     id: user._id,
+                     title: user.title
+                   })
+                 })
+                 res.status(200).send(users)
+               }
+             })
+       }
+       else{
+         res.status(401).json({detail: "Invalid access!!"})
+       }
+     }
+   })
+ }
+ else{
+   res.status(401).json({detail: "No such user found!!"})
+ }
 };
 
 const createToDo = async (req, res) => {
   // Check for the token and create a todo
   // or throw error correspondingly
+  const token = verification(req,res);
+  if(token != false){
+    Token.findOne({token: token}, function(err, foundToken){
+      if(err)
+        res.send(err);
+      else{
+        if(foundToken){
+              const newTodo = new ToDo({
+                               title: req.body.title,
+                               createdBy: foundToken.user});
+              newTodo.save(function(err){
+                if(!err)
+                  res.status(200).json({id: req.params.id, title: newTodo.title})
+                else
+                  res.send("Cannot create todo!!!");
+              })
+        }
+        else
+          res.status(401).json({detail: "Invalid access!!"})
+      }
+    })
+  }
+  else
+    res.status(401).json({detail: "No such user found!!"})
 };
 
 const getParticularToDo = async (req, res) => {
```

Comment:
I am able to access todo using other tokens. A major flaw. Seems middleware would need update.